File: libraries/typescript/src/stt/gemini.ts
/** Gemini multimodal STT for Patter pipeline mode. */
import { GeminiSTT as _GeminiSTT } from "../providers/gemini-stt";

/** Constructor options for the Gemini `STT` adapter. */
export interface GeminiSTTOptions {
  /** API key. Falls back to GEMINI_API_KEY / GOOGLE_API_KEY env var when omitted. */
  apiKey?: string;
  /** Multimodal model (default `gemini-2.5-flash`). */
  model?: string;
  /** Inbound PCM rate from the pipeline (default 16000). */
  sampleRate?: number;
}

/**
 * Gemini multimodal STT (`gemini-2.5-flash`) — transcribes AND tags the
 * caller's vocal tone (`[tone: ...]`) so the agent can mirror mood.
 *
 * @example
 * ```ts
 * import * as gemini from "getpatter/stt/gemini";
 * const stt = new gemini.STT();              // reads GEMINI_API_KEY
 * ```
 */
export class STT extends _GeminiSTT {
  static readonly providerKey = "gemini_stt";
  constructor(opts: GeminiSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.GEMINI_API_KEY ?? process.env.GOOGLE_API_KEY;
    if (!key) {
      throw new Error(
        "Gemini STT requires an apiKey. Pass { apiKey: '...' } or set " +
          "GEMINI_API_KEY / GOOGLE_API_KEY in the environment.",
      );
    }
    super(key, opts.model ?? "gemini-2.5-flash", opts.sampleRate ?? 16000);
  }
}

File: libraries/python/getpatter/stt/gemini.py
"""Gemini multimodal STT for Patter pipeline mode (Beta)."""

from __future__ import annotations

from typing import ClassVar

from getpatter.providers.gemini_stt import (
    GEMINI_STT_DEFAULT_MODEL,
    GEMINI_STT_DEFAULT_SAMPLE_RATE,
    GeminiSTT as _GeminiSTT,
)

__all__ = ["STT"]


class STT(_GeminiSTT):
    """Gemini multimodal STT (``gemini-2.5-flash``) (Beta).

    Transcribes AND tags the caller's vocal tone (``[tone: ...]``) so the agent
    can mirror mood.

    Example::

        from getpatter.stt import gemini

        stt = gemini.STT()                      # reads GEMINI_API_KEY
        stt = gemini.STT(api_key="...", model="gemini-2.5-flash")

    One request per turn, fired at VAD speech-end: there are no interim
    partials. Prefer Deepgram / Soniox / AssemblyAI when turn latency matters
    more than tone.
    """

    provider_key: ClassVar[str] = "gemini_stt"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        model: str = GEMINI_STT_DEFAULT_MODEL,
        sample_rate: int = GEMINI_STT_DEFAULT_SAMPLE_RATE,
    ) -> None:
        super().__init__(api_key=api_key, model=model, sample_rate=sample_rate)
